app/controllers/network_routes_controller.go

```diff
@@ -680,6 +692,23 @@ func (nrc *NetworkRoutingController) AdvertiseClusterIp(clusterIp string) error
 	return nil
 }
 
+// UnadvertiseClusterIP  unadvertises the service cluster ip
+func (nrc *NetworkRoutingController) UnadvertiseClusterIp(clusterIp string) error {
+	glog.V(2).Infof("Unadvertising route: '%s/%s via %s' to peers", clusterIp, strconv.Itoa(32), nrc.nodeIP.String())
+
+	pathList := []*table.Path{table.NewPath(nil, bgp.NewIPAddrPrefix(uint8(32),
+		clusterIp), true, nil, time.Now(), false)}
+
+	err := nrc.bgpServer.DeletePath([]byte(nil), 0, "", pathList)
+
+	if err != nil {
+		return fmt.Errorf(err.Error())
```

Comment:
just 
```go
return err 
```

Reply:
I agree with you, but the AdvertiseClusterIp method above it should be refactored as well then, since the methods are structured the same

Reply:
feel free to fix the method above too :)

Reply:
alrighty then :)